src/query/crypt_opfunc.c

```diff
@@ -697,3 +699,359 @@ crypt_generate_random_bytes (char *dest, int length)
 
   return NO_ERROR;
 }
+
+// 
+// *INDENT-OFF*
+static struct
+{
+  unsigned char nonce[16];              // See TDE_DK_NONCE_LENGTH in tde.h
+  unsigned char master_key[TDE_MASTER_KEY_LENGTH];
+} evp_cipher = { {0, }, {0, } };  // Do not omit the this initialization settings.
+// *INDENT-ON*
+
+static int
+init_dblink_cipher (EVP_CIPHER_CTX ** ctx, const EVP_CIPHER ** cipher_type, bool is_aes_algorithm)
+{
+  static int is_init_done = 0;
+
+  if (is_init_done == 0)
+    {
+      memset (evp_cipher.nonce, 0x07, sizeof (evp_cipher.nonce));
+
+#if !defined(CS_MODE)
+      memcpy (evp_cipher.master_key, tde_Cipher.data_keys.log_key, TDE_MASTER_KEY_LENGTH);
+#endif
+      is_init_done = 1;
+    }
+
+#if defined(CS_MODE)
+  if (tde_Cipher.is_loaded == false)
+    {
+      int err;
+      extern int tde_get_data_keys ();	// declared in "network_interface_cl.c"
+      if ((err = tde_get_data_keys ()) != NO_ERROR)
+	{
+	  return err;
+	}
+      tde_Cipher.is_loaded = true;
+    }
+  memcpy (evp_cipher.master_key, tde_Cipher.data_keys.log_key, TDE_MASTER_KEY_LENGTH);
+#endif
+
```

Comment:
```suggestion
memcpy (evp_cipher.master_key, tde_Cipher.data_keys.log_key, TDE_MASTER_KEY_LENGTH);
```
it can be sperated from CS_MODE definition together with prior line #722

Reply:
In case of CS_MODE, the value is copied every time,
In case of SA_MODE(= not CS_MODE), it is copied only once initially.

Reply:
Ok, I understood.